mage_ai/orchestration/execution_process_manager.py

```diff
@@ -90,3 +92,152 @@ def clean_up_processes(self, include_child_processes=True):
 
 
 execution_process_manager = ExecutionProcessManager()
+
+
+def start_session_and_run(*target_args):
+    from mage_ai.orchestration.db import db_connection
+
+    if len(target_args) == 0:
+        return None
+    target = target_args[0]
+    args = target_args[1:]
+
+    db_connection.start_session(force=True)
+
+    try:
+        results = target(*args)
+    finally:
+        db_connection.close_session()
+    return results
+
+
+def create_process(target, args=()):
+    from mage_ai.orchestration.db import engine 
+
+    new_args = [target, *args]
+    engine.dispose()
+    return mp.Process(target=start_session_and_run, args=new_args)
+
+
+class Worker(mp.Process):
+    def __init__(self, queue: mp.Queue):
+        super().__init__()
+        self.queue = queue
+        self.job_uuid = None
+
+    def run(self):
+        while True:
+            if not self.queue.empty():
+                args = self.queue.get()
+
+                self.job_uuid = args[0]
+                run_args = args[1:]
+
+                try:
+                    start_session_and_run(*run_args)
+                finally:
+                    self.job_uuid = None
```

Comment:
If there're exceptions, the process will be terminated. Setting job_uuid does nothing.

Reply:
if the process dies, we'll need to create a new one.

Reply:
oh good point